homeassistant/helpers/template.py

```diff
@@ -523,16 +559,29 @@ def expand(hass: HomeAssistantType, *args: Any) -> Iterable[State]:
             # ignore other types
             continue
 
-        # pylint: disable=import-outside-toplevel
-        from homeassistant.components import group
-
-        if split_entity_id(entity_id)[0] == group.DOMAIN:
-            # Collect state will be called in here since it's wrapped
-            group_entities = entity.attributes.get(ATTR_ENTITY_ID)
-            if group_entities:
-                search += group_entities
+        if entity_id.startswith(_GROUP_DOMAIN_PREFIX):
+            # Collect state would be called in here since it's wrapped
+            # so we call hass.states.get since we only want to collect
+            # the underlying entity ids.
+            #
+            # We only want to look at the underlying entities because
+            # when a expand() is in the template we check each entity
+            # instead of the state of the group. If we were to include
+            # the group we would end up re-rendering the template again
+            # when the group had finished processing the state change of
+            # the underlying entities.
```

Comment:
What if the group members change?

Reply:
I’ll write tests for this and adjust as needed

Reply:
Looks like we are stuck with tracking both unless we want to listen for reloads which drives up the complexity so I went ahead and reverted the change and added tests to cover the reload case.